### PR TITLE
Platform incompatibilities

### PR DESCRIPTION
**Pull request by sebn**

### CODE REVIEW

File: gui/app/main.js [hunk 1]
```diff
@@ -318,6 +333,47 @@ const chooseSyncPath = () => {
   sendToMainWindow('registration-done')
 }
 
+const incompatibilitiesErrorMessage = (i) => {
+  const reasons = []
+  const docType = translate(`Helpers ${i.docType}`)
+  if (i.reservedChars) {
+    reasons.push(
+      interpolate(
+        translate('Error {0} names cannot include « {1} » characters'),
```
Comment (nono) [inline gui/app/main.js]: I think that « ... » is used in french, but not in english, where “...” is prefered

File: src/utils/regexp.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* @flow */
+
+// Same as String#match, but return all the matches
```
Comment (nono) [inline src/utils/regexp.js]: Isn't the `g` flag on the regexp made for that?

```js
string.match(/[aeiou]/g)
```

File: gui/app/main.js [hunk 3]
```diff
@@ -352,6 +408,12 @@ const startSync = (force) => {
       addFile(info)
       removeFile(old)
     })
+    desktop.events.on('platform-incompatibilities', incompatibilities => {
+      // TODO: Index incompatibilities to prevent notification overload
+      incompatibilities.forEach(i => {
+        sendErrorToMainWindow(incompatibilitiesErrorMessage(i))
```
Comment (nono) [inline gui/app/main.js]: A debounce on `sendErrorToMainWindow` is a good idea for your TODO

File: src/remote/watcher.js [hunk 4]
```diff
@@ -132,6 +135,11 @@ export default class RemoteWatcher {
     let doc: Metadata = conversion.createMetadata(remote)
     const docType = doc.docType
     ensureValidPath(doc)
+    const incompatibilities = pathPlatformIncompatibilities(doc)
+    if (incompatibilities) {
+      this.events.emit('platform-incompatibilities', incompatibilities)
+      return
+    }
```
Comment (nono) [inline src/remote/watcher.js]: I'd have put this check in `Prep`, but it's OK for here

File: src/utils/regexp.js (hunk 2 above)
Comment (sebn) [inline src/utils/regexp.js]: I believe I generalized it without using other match args anymore. Thanks!

**Comment by sebn** [comment]
Added fixes according to feedback. I'll merge as soon as build is green.

**Comment by sebn** [comment]
@nono Do you think we should fix the debounce issue right now?

I suppose we could just add and use `lodash.debounce`.

**Comment by nono** [comment]
@sebn it's probably easy to do, so better to do it now

**Comment by sebn** [comment]
:ship: ?

**Pull request merged**